test/extended/apiserver/tls.go

```diff
@@ -1,31 +1,467 @@
+				time.Sleep(backoff)
+				continue
+			}
+			return fmt.Errorf("failed to start oc port-forward command: %w", err)
+		}
+
+		scanner := bufio.NewScanner(stdout)
+		if scanner.Scan() {
+			e2e.Logf("oc port-forward output: %s", scanner.Text())
+
+			toExecute(localPort)
+
+			cmd.Process.Kill()
+			cmd.Wait()
+
+			return nil
+		}
+
+		if err := scanner.Err(); err != nil {
+			if attempt < maxConnectRetries-1 {
+				backoff := initialBackoff * time.Duration(math.Pow(2, float64(attempt)))
+				time.Sleep(backoff)
+				continue
+			}
+			return fmt.Errorf("failed to read oc port-forward output: %w", err)
+		}
+
+		// the port-forward failed to start properly
+		if attempt < maxConnectRetries-1 {
+			backoff := initialBackoff * time.Duration(math.Pow(2, float64(attempt)))
+			time.Sleep(backoff)
+		}
+	}
+
+	return fmt.Errorf("port forwarding failed after %d attempts", maxConnectRetries)
+}
+
+func getIPFamilyForCluster(client exutil.CLI, namespace string) IPFamily {
+	podIPs, err := createPod(client, namespace, "test-ip-family-pod")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return getIPFamily(podIPs)
+}
+
+func getIPFamily(podIPs []corev1.PodIP) IPFamily {
+	switch len(podIPs) {
+	case 1:
+		ip := net.ParseIP(podIPs[0].IP)
+		if ip.To4() != nil {
+			return IPv4
+		} else {
+			return IPv6
+		}
+	case 2:
+		ip1 := net.ParseIP(podIPs[0].IP)
+		ip2 := net.ParseIP(podIPs[1].IP)
+		if ip1 == nil || ip2 == nil {
+			return Unknown
+		}
+		if (ip1.To4() == nil) == (ip2.To4() == nil) {
+			return Unknown
+		}
+		return DualStack
+	default:
+		return Unknown
+	}
+}
+
+func createPod(client exutil.CLI, ns, generateName string) ([]corev1.PodIP, error) {
+	pod := frameworkpod.NewAgnhostPod(ns, "", nil, nil, nil)
+	pod.ObjectMeta.GenerateName = generateName
+	pod.Spec.SecurityContext = &corev1.PodSecurityContext{
+		RunAsNonRoot: ptr.To(true),
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+	}
+	pod.Spec.Containers[0].SecurityContext = &corev1.SecurityContext{
+		AllowPrivilegeEscalation: ptr.To(false),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		},
+		ReadOnlyRootFilesystem: ptr.To(true),
+		Privileged:             ptr.To(false),
+	}
+
+	execPod, err := client.AdminKubeClient().CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	var podIPs []corev1.PodIP
+	err = wait.PollImmediate(poll, 2*time.Minute, func() (bool, error) {
+		retrievedPod, err := client.AdminKubeClient().CoreV1().Pods(execPod.Namespace).Get(context.TODO(), execPod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		podIPs = retrievedPod.Status.PodIPs
+		return retrievedPod.Status.Phase == corev1.PodRunning, nil
+	})
+	return podIPs, err
+}
+
 var _ = g.Describe("[sig-api-machinery][Feature:APIServer]", func() {
 	defer g.GinkgoRecover()
 
 	oc := exutil.NewCLI("apiserver")
 
 	g.It("TestTLSDefaults", func() {
```

Comment:
how does this test differs from the one that has been added ?

Reply:
The older test goes deeper and tests acceptable cipher suites, but it only checks the API server, and only when it's running under the default configuration.